docs/devguide/xtypes.rst

```diff
@@ -1315,10 +1331,135 @@ They provides the same operations as any other DataWriter or DataReader, but wit
 See :ref:`xtypes--populating-data-samples-with-dynamicdata` for details on creating DynamicData objects for use with the DynamicDataWriter interface.
 See :ref:`xtypes--interpreting-data-samples-with-dynamicdata` for details on using DynamicData objects obtained from the DynamicDataReader interface.
 
+.. _xtypes-dynamicdata-and-idl-generated-types:
+
+``DynamicData`` and IDL-Generated Types
+=======================================
+
+``DynamicData`` allows writing code that is much more generic than using the IDL-generated types.
+There are still reasons to use the IDL-generated types though.
+They are more convenient to use and are more likely to be used in existing code.
+You can still make use of the flexibility of ``DynamicData`` though by either interchanging with or wrapping IDL-generated types.
+
+Interchanging Data using ``create_sample`` and ``create_dynamic_sample``
+------------------------------------------------------------------------
+
+.. versionadded:: 3.29.0
+
+The XTypes specification defines ``create_sample`` and ``create_dynamic_sample`` on the ``TypeSupport`` for every :term:`topic type` to interchange ``DynamicData`` and IDL-generated types.
+In addition to this, OpenDDS adds ``create_sample_rc`` and ``create_dynamic_sample_rc`` to indicate failure and provide a more consistent interface.
+The standard methods will only warn if there was an error and can't indicate that an error occured.
+
+``create_sample`` and ``create_sample_rc``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``create_sample`` takes a ``DynamicData`` and returns the IDL-generated type.
+One thing to note about this method is the C++ to IDL specification requires that returns like this can be either returned on the stack or the heap depending on if they are :omgspec:`"variable-length" <cpp03:5.11 Mapping for Structured Types>`.
+For example if there's a topic type ``Foo`` that doesn't contain a ``vector`` or ``string``, then:
+
+.. code-block:: omg-idl
+
+  struct Foo {
+    int32 bar;
+  };
+
+.. code-block:: cpp
+
+  FooTypeSupport_var type_support = new FooTypeSupportImpl();
+  DynamicData_var data = /* ... */;
+
+  Foo foo = type_support->create_sample(data);
+
+If the type contains a ``vector`` or ``string`` anywhere, then then it returns a pointer that's meant to go inside a ``_var`` class:
```

Comment:
```suggestion
If the type contains a ``sequence`` or ``string`` anywhere, then then it returns a pointer that's meant to go inside a ``_var`` class:
```
Are there other IDL constructs that can be variable-length and can appear in a DDS topic type?

Reply:
If everything in the specs, `tao_idl`, and OpenDDS was ideal and consistent, then probably `map`. [Everything else](https://www.omg.org/spec/CPP/1.3/PDF#G6.69275) doesn't look relevant to DDS.

Reply:
How about putting a comment in the rst (if it supports comments) that map should be included here when it's ready?